File: XMLCheck/Handler/CheckRulesXML06/Xml6RuleBase.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML06
{
    public abstract class Xml6RuleBase:IXml6Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml6Model model,HoSoContext context);

        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml6/xml6RuleBase.ts
import { Xml6Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { BaseRule } from '../base.rule';

export interface IXml6Rule {
  readonly key: string;
  readonly isEnabled: boolean;
  check(model: Xml6Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml6RuleBase extends BaseRule<Xml6Model> implements IXml6Rule {}
